src/io/eval.rs

```diff
@@ -311,66 +185,140 @@ where
         obf_params.public_circuit,
     );
     log_mem("final_circuit built");
-    let last_input_encodings = encodings_cur;
-    let output_encodings = final_circuit.eval::<BggEncoding<M>>(
-        &params,
-        &last_input_encodings[0],
-        &last_input_encodings[1..],
-    );
-    log_mem("final_circuit evaluated");
-    let output_encoding_ints = output_encodings
-        .par_chunks(log_base_q)
-        .map(|digits| BggEncoding::digits_to_int(digits, &params))
-        .collect::<Vec<_>>();
-    let output_encodings_vec = output_encoding_ints[0].concat_vector(&output_encoding_ints[1..]);
-    log_mem("final_circuit evaluated and recomposed");
-    let final_preimage = M::read_from_files(
+    let final_preimage_f = M::read_from_files(
         &obf_params.params,
         m_b,
         packed_output_size,
         &dir_path,
-        "final_preimage",
+        "final_preimage_f",
     );
     log_mem("final_preimage loaded");
-    let final_v = p_cur * final_preimage;
+    // v := p * K_F
+    let final_v = p_cur.clone() * final_preimage_f;
     log_mem("final_v computed");
-    let z = output_encodings_vec - final_v;
-    log_mem("z computed");
-    debug_assert_eq!(z.size(), (1, packed_output_size));
+
     #[cfg(feature = "debug")]
     if obf_params.encoding_sigma == 0.0 &&
         obf_params.hardcoded_key_sigma == 0.0 &&
         obf_params.p_sigma == 0.0
     {
-        let hardcoded_key = <<M as PolyMatrix>::P as Poly>::read_from_file(
```

Comment:
I think we can still implement these two debug assertion that you commented out. If you need the key `minus_t_bar` to compute the expected encoding you can store it from the obfuscation only in debug mode